kitsune/search/views.py

```diff
@@ -74,38 +156,31 @@ def simple_search(request, template=None):
         new_url = reverse('search.advanced') + '?' + request.GET.urlencode()
         return HttpResponseRedirect(new_url)
 
-    # JSON-specific variables
-    is_json = (request.GET.get('format') == 'json')
-    callback = request.GET.get('callback', '').strip()
-    content_type = 'application/x-javascript' if callback else 'application/json'
+    request = json_markup_request(request)
 
     # Check callback is valid
-    if is_json and callback and not jsonp_is_valid(callback):
+    if request.JSON_CALLBACK and not jsonp_is_valid(request.JSON_CALLBACK):
         return HttpResponse(
             json.dumps({'error': _('Invalid callback function.')}),
-            content_type=content_type,
+            content_type=request.CONTENT_TYPE,
             status=400)
 
     search_form = SimpleSearchForm(request.GET, auto_id=False)
 
     if not search_form.is_valid():
-        if is_json:
+        if request.IS_JSON:
             return HttpResponse(
                 json.dumps({'error': _('Invalid search data.')}),
-                content_type=content_type,
+                content_type=request.CONTENT_TYPE,
```

Comment:
Looking at this, I don't think it has ever correctly dealt with jsonp callbacks when there is an error. Do we even use those?